### PR TITLE
Inline CSS background images directly into the CSS

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
A nice advantage of this is that it removes a few entries in the list of static files.

r? @notriddle

### CODE REVIEW

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -806,8 +808,8 @@ so that we can apply CSS-filters to change the arrow color in themes */
 	background-repeat: no-repeat;
 	background-size: 20px;
 	background-position: calc(100% - 2px) 56%;
-	/* image is black color */
-	background-image: url("down-arrow-927217e04c7463ac.svg");
+	/* down arrow (image is black color) */
+	background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiBpZD0iTGF5ZXJfMSIgd2lkdGg9IjEyOCIgaGVpZ2h0PSIxMjgiIGVuYWJsZS1iYWNrZ3JvdW5kPSJuZXcgMCAwIDEyOCAxMjgiIHZlcnNpb249IjEuMSIgdmlld0JveD0iLTMwIC0yMCAxNzYgMTc2IiB4bWw6c3BhY2U9InByZXNlcnZlIj48Zz48bGluZSB4MT0iMTExIiB4Mj0iNjQiIHkxPSI0MC41IiB5Mj0iODcuNDk5IiBmaWxsPSJub25lIiBzdHJva2U9IiMwMDAwMDAiIHN0cm9rZS1saW5lY2FwPSJzcXVhcmUiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgc3Ryb2tlLXdpZHRoPSIxMiIvPjxsaW5lIHgxPSI2NCIgeDI9IjE3IiB5MT0iODcuNDk5IiB5Mj0iNDAuNSIgZmlsbD0ibm9uZSIgc3Ryb2tlPSIjMDAwMDAwIiBzdHJva2UtbGluZWNhcD0ic3F1YXJlIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIHN0cm9rZS13aWR0aD0iMTIiLz48L2c+PC9zdmc+");
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:812]: If we inline it like in #104267, no need to worry about doing the inlining in base64 (unless it's the whole inlining itself that bothers you?).

Also agreed, with content not in base64, backslash looks much better so I'll use it instead.

**Comment by GuillaumeGomez** [comment]
Updated to keep the svg content as is instead of using base64 and added some backslashes.

**Review by notriddle — CHANGES_REQUESTED** [review]

File: src/librustdoc/html/static/css/rustdoc.css [hunk 2]
```diff
@@ -1518,7 +1526,10 @@ details.toggle[open] > summary.hideme > span {
 }
 
 details.toggle[open] > summary::before {
-	background: url("toggle-minus-31bbd6e4c77f5c96.svg") no-repeat top left;
+	/* toggle minus */
+	background: url('data:image/svg+xml;utf8,<svg width="17" height="17" \
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1530]: Good point.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 3]
```diff
@@ -806,8 +806,13 @@ so that we can apply CSS-filters to change the arrow color in themes */
...
 	background-position: calc(100% - 2px) 56%;
-	/* image is black color */
-	background-image: url("down-arrow-927217e04c7463ac.svg");
+	/* down arrow (image is black color) */
+	background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" \
+width="128" height="128" enable-background="new 0 0 128 128" version="1.1" \
+viewBox="-30 -20 176 176" xml:space="preserve"><g><line x1="111" x2="64" y1="40.5" y2="87.499" \
+fill="none" stroke="black" stroke-linecap="square" stroke-miterlimit="10" stroke-width="12"/>\
+<line x1="64" x2="17" y1="87.499" y2="40.5" fill="none" stroke="black" stroke-linecap="square" \
+stroke-miterlimit="10" stroke-width="12"/></g></svg>');
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:815]: Wow that's much smaller. Did you use a software in particular to generate the smaller version?

**Comment by GuillaumeGomez** [comment]
Updated!

File: src/librustdoc/html/static/css/rustdoc.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css]: The famous `notriddle` software. ;)

**Review by notriddle — APPROVED** [review]

**Pull request merged**